### PR TITLE
Different messages for different values of NewlineAtEndOfFile added in SA1518CodeMustNotContainBlankLinesAtEndOfFile

### LINKED ISSUE #1847: SA1518 message misleading when newlineAtEndOfFile is set to require
##### stylecop.json:

```json
{
  "settings": {
    "layoutRules": {
	  "newlineAtEndOfFile": "require"
    }
  }
}
```

##### Code file:
```cs
namespace SomeNamespace
{
    public class SomeClass
    {
    }
} // <--- No new line here
```

##### Analyzer warning:
    Warning	SA1518	Code must not contain blank lines at end of file

**In fact my code SHOULD contain blank line.**

P.S. thanks for implementing this setting :)

### PR DESCRIPTION
**Pull request by NikolayIT**
Fixes #1847

### CODE REVIEW

**Comment by NikolayIT** [comment]
I need your assistance here. :)
As expected few unit tests failed after changing rule messages. In order to fix them I am thinking of the following approach:
Add a new `DiagnosticVerifier` method to compare diagnostics ignoring their messages and use it in unit tests.
Is this OK?

File: StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518CodeMustNotContainBlankLinesAtEndOfFile.cs [hunk 1]
```diff
@@ -28,20 +28,30 @@ internal class SA1518CodeMustNotContainBlankLinesAtEndOfFile : DiagnosticAnalyze
         /// The ID for diagnostics produced by the <see cref="SA1518CodeMustNotContainBlankLinesAtEndOfFile"/> analyzer.
         /// </summary>
         public const string DiagnosticId = "SA1518";
-        private const string Title = "Code must not contain blank lines at end of file";
-        private const string MessageFormat = "Code must not contain blank lines at end of file";
-        private const string Description = "The code file has blank lines at the end.";
+        private const string TitleForAllowSetting = "Code must not contain blank lines at end of file";
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518CodeMustNotContainBlankLinesAtEndOfFile.cs]: :exclamation: The title appears in the rule set editor, so it needs to be a fixed value. Only the message and description can change.

Comment (NikolayIT) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518CodeMustNotContainBlankLinesAtEndOfFile.cs]: (y)

**Comment by sharwell** [comment]
> As expected few unit tests failed after changing rule messages.

There is an overload of `CSharpDiagnostic` which takes a `DiagnosticDescriptor`. For example, take a look at the difference between the `SA1200OutsideNamespaceUnitTests` and `SA1200UnitTests`.

**Comment by NikolayIT** [comment]
@sharwell, I think I get it. Is this solution convenient?

File: StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518CodeMustNotContainBlankLinesAtEndOfFile.cs [hunk 2]
```diff
@@ -28,20 +28,35 @@ internal class SA1518CodeMustNotContainBlankLinesAtEndOfFile : DiagnosticAnalyze
         /// The ID for diagnostics produced by the <see cref="SA1518CodeMustNotContainBlankLinesAtEndOfFile"/> analyzer.
         /// </summary>
         public const string DiagnosticId = "SA1518";
-        private const string Title = "Code must not contain blank lines at end of file";
-        private const string MessageFormat = "Code must not contain blank lines at end of file";
-        private const string Description = "The code file has blank lines at the end.";
+
+        private const string Title = "Fix blank lines at the end of the file";
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518CodeMustNotContainBlankLinesAtEndOfFile.cs]: :thought_balloon: What about this?

> Use line endings correctly at end of file

**Comment by sharwell** [comment]
Alright now it looks good to me. I'm going to send you a pull request (to this branch in your fork) to clean up a couple super minor details. When you merge my pull request, it will automatically update this pull request to include everything. Then I can merge this pull request :+1:

**Comment by NikolayIT** [comment]
Deal :joy_cat:

**Comment by sharwell** [comment]
@NikolayIT I sent pull request NikolayIT/StyleCopAnalyzers#1. :smile:

**Pull request merged**